homeassistant/components/kostal_plenticore/select.py

```diff
@@ -105,24 +102,18 @@ def __init__(
         description: PlenticoreSelectEntityDescription,
         entry_id: str,
         platform_name: str,
-        device_class: str | None,
-        current_option: str | None,
         device_info: DeviceInfo,
     ) -> None:
         """Create a new Select Entity for Plenticore process data."""
         super().__init__(coordinator)
         self.entity_description = description
         self.entry_id = entry_id
         self.platform_name = platform_name
-        self._attr_device_class = device_class
         self.module_id = description.module_id
         self.data_id = description.key
         self._attr_options = description.options
-        self.all_options = description.options
-        self._attr_current_option = current_option
-        self._is_on = description.is_on
+        self._attr_current_option = description.current_option
```

Comment:
I see now that this won't work. The entity overrides the `current_option` property.

Reply:
You're right!
Done